pkg/columns/formatter/json/json.go

```diff
@@ -35,7 +35,7 @@ type column[T any] struct {
 type Formatter[T any] struct {
 	options *Options
 	columns []*column[T]
-	printer func(entry *T, buf *encodeState)
+	printer func(indent string, entry *T, buf *encodeState)
```

Comment:
nit: `indent` feels like the least important parameter, so maybe it'd be better placed at the end. Seeing this, I'd also suggest to put `buf` first, as most times the target one writes to is the first parameter - but feel free to ignore.